lib/docker-image-tree.coffee

```diff
@@ -32,6 +36,11 @@ exports.annotateTree = annotateTree = (layer_mtimes, tree) ->
 
 docker = new Docker(socketPath: '/var/run/docker.sock')
 docker = Promise.promisifyAll(docker)
+# Hack dockerode to promisify internal classes' prototypes
+Promise.promisifyAll(Docker({}).getImage().constructor.prototype)
+
+exports.getDocker = ->
```

Comment:
if you want to wrap the docker module in its own local module that adds promises can you extract it into its own file?